tests/planner/parabolic/test_ParabolicSmoother.cpp

```diff
@@ -0,0 +1,240 @@
+#include <aikido/constraint/Satisfied.hpp>
+
+using Eigen::Vector2d;
+using Eigen::Vector3d;
+using aikido::trajectory::Interpolated;
+using aikido::statespace::GeodesicInterpolator;
+using aikido::statespace::R2;
+using aikido::statespace::CartesianProduct;
+using aikido::statespace::SO2;
+using aikido::statespace::SO3;
+using aikido::statespace::StateSpacePtr;
+using aikido::constraint::Satisfied;
+using aikido::planner::parabolic::convertToSpline;
+using aikido::planner::parabolic::doShortcut;
+using aikido::planner::parabolic::doBlend;
+using aikido::planner::parabolic::doShortcutAndBlend;
+
+class ParabolicSmootherTests : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    mStateSpace = std::make_shared<R2>();
+    mMaxVelocity = Eigen::Vector2d(1., 1.);
+    mMaxAcceleration = Eigen::Vector2d(2., 2.);
+
+    mInterpolator = std::make_shared<GeodesicInterpolator>(mStateSpace);
+    mStraightLine = std::make_shared<Interpolated>(
+      mStateSpace, mInterpolator);
+
+    auto state = mStateSpace->createState();
+
+    state.setValue(Vector2d(1., 2.));
+    mStraightLine->addWaypoint(0., state);
+
+    state.setValue(Vector2d(3., 4.));
+    mStraightLine->addWaypoint(1., state);
+
+  }
+
+  std::shared_ptr<R2> mStateSpace;
+  Eigen::Vector2d mMaxVelocity;
+  Eigen::Vector2d mMaxAcceleration;
+
+  std::shared_ptr<GeodesicInterpolator> mInterpolator;
+  std::shared_ptr<Interpolated> mStraightLine;
+};
+
+TEST_F(ParabolicSmootherTests, convertInterpolatedToSpline)
+{
+    auto spline = convertToSpline(*mStraightLine.get());
+
+    auto splineState = mStateSpace->createState();
+    auto interpolatedState = mStateSpace->createState();
+    Eigen::VectorXd splineVec, interpolatedVec;
+    Eigen::VectorXd splineTangent, interpolatedTangent;
+    for(double t=spline->getStartTime();t<=spline->getEndTime();t+=1e-2)
+    {
+        spline->evaluate(t, splineState);
+        mStraightLine->evaluate(t, interpolatedState);
+        mStateSpace->logMap(splineState, splineVec);
+        mStateSpace->logMap(interpolatedState, interpolatedVec);
+        EXPECT_TRUE(interpolatedVec.isApprox(interpolatedVec));
+
+        spline->evaluateDerivative(t, 1, splineTangent);
+        mStraightLine->evaluateDerivative(t, 1, interpolatedTangent);
+        EXPECT_TRUE(splineTangent.isApprox(interpolatedTangent));
+    }
+}
+
+TEST_F(ParabolicSmootherTests, doShort)
+{
+   Interpolated nonStraightLine(mStateSpace, mInterpolator);
+
+   auto state = mStateSpace->createState();
+   Vector2d p1(1., 1.), p2(1., 2.), p3(2., 2.),
+           p4(2., 3.), p5(3., 3.);
+   state.setValue(p1);
+   nonStraightLine.addWaypoint(0., state);
+   state.setValue(p2);
+   nonStraightLine.addWaypoint(0.25, state);
+   state.setValue(p3);
+   nonStraightLine.addWaypoint(0.5, state);
+   state.setValue(p4);
+   nonStraightLine.addWaypoint(0.75, state);
+   state.setValue(p5);
+   nonStraightLine.addWaypoint(1., state);
```

Comment:
No.... In Hauser's code, shortcut will be executed only when the waypoint number is larger than 3....